sklearn/tree/_criterion.pyx

```diff
@@ -962,6 +963,108 @@ cdef class MSE(RegressionCriterion):
         impurity_left[0] /= self.n_outputs
         impurity_right[0] /= self.n_outputs
 
+cdef class MAE(RegressionCriterion):
+    """Mean absolute error impurity criterion
+    """
+    cdef void node_value(self, double* dest) nogil:
+        """Computes the node value of samples[start:end] into dest."""
+        with gil:
+            print "entered node_value"
+        cdef double* sample_weight = self.sample_weight
+        cdef SIZE_t* samples = self.samples
+        
+        cdef DOUBLE_t* y = self.y
+        cdef SIZE_t start = self.start
+        cdef SIZE_t end = self.end
+        cdef SIZE_t i
+        cdef SIZE_t p
+        cdef SIZE_t k
+        cdef DOUBLE_t w = 1.0
+        cdef DOUBLE_t y_ik
+
+        cdef DOUBLE_t sum_weights = 0.0
+        cdef SIZE_t median_index = 0
+        cdef DOUBLE_t Sum
+
+        y_vals = NULL
+        weights = NULL
+        y_vals = <double*> calloc(self.n_node_samples, sizeof(double))
```

Comment:
Not sure if this is the correct way to instantiate some sort of collection object in cython...